src/common/helpers/snap-builds.js

```diff
@@ -136,7 +138,7 @@ const LaunchpadStoreUploadStates = {
 function mapBuildAndUploadStates(buildState, uploadState) {
   switch (buildState) {
     case LaunchpadBuildStates.NEEDSBUILT:
-      return UserFacingState.NEVER_BUILT;
+      return UserFacingState.BUILDING_SOON;
```

Comment:
When there is build object on LP it means that it will be building soon for BSI UI.